### ISSUE TITLE
[MCHECKSTYLE-49] Review and revise plugin documentation

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-checkstyle-plugin#407]
**[Maria Odea B. Ching](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=oching)** opened **[MCHECKSTYLE-49](https://issues.apache.org/jira/browse/MCHECKSTYLE-49?redirect=false)** and commented

---

**Attachments:**
- [documentation.patch](https://issues.apache.org/jira/secure/attachment/12714721/documentation.patch) (_9.27 kB_)

**Comment by jira-importer** [comment]
**[Maria Odea B. Ching](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=oching)** commented

Review comments by Stephen Duncan (from Maven Dev List):

---

On the "Multimodule Configuration" documentation:

As I just mentioned on a question on the user's list, I don't think
it's correct to specify the "build-tools" dependency as a dependency
of the plugin.  While this will work if you manually install the
build-tools jar, it will not download it from an internal repository.
It should instead be specified as build extension like in the "Using
Custom Developed Chechstyle Check Modules" example.  (Also not the
spelling mistake in that title: ChecHstyle).

Because this is somewhat confusing, I think it should mentioned either
in the "Using a Custom Checkstyle Checker Configuration" as a way of
using a classpath reference, or it should be it's own guide on using a
shared jar for configuration.

- Stephen

**Comment by jira-importer** [comment]
**[Maria Odea B. Ching](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=oching)** commented

I've already revised and applied the review comments in the plugin docs.

Btw, for the shared jar configuration issue in "Multimodule Configuration", I've just revised the page and mentioned the shared jar configuration issue (declaring it as an extension instead of as a plugin dependency) instead of having a separate guide for that. I did this because I think it would just have almost the same example used in "Multimodule Configuration".

The staging site (http://people.apache.org/~oching/maven-checkstyle-plugin) has already been updated. Thanks :)
